modules/core/include/opencv2/core/quaternion.hpp

```diff
@@ -0,0 +1,1138 @@
+     *
+     * For example
+     * ```
+     * Quatf q(1,2,3,4);
+     * q.sqrt();
+     *
+     * q = {1,0,0,0};
+     * q.sqrt(true); //true means q is a unit quaternion
+     * ```
+     */
+    Quat<_Tp> sqrt(bool assumeUnit=false) const;
+
+    /**
+     * @brief return the value of power function with quaternion \f$q\f$.
+     * \f[p^q = e^{q\ln(p)}\f].
+     * @param p base quaternion of power function.
+     * @param q index quaternion of power function.
+     * @param assumeUnit if true, quaternion \f$p\f$ should be a unit quaternion and this function will save some computations.
+     *
+     * For example
+     * ```
+     * Quatd p(1,2,3,4);
+     * Quatd q(5,6,7,8);
+     * power(p, q);
+     *
+     * p = p.normalize();
+     * power(p, q, true); //true means p is a unit quaternion
+     * ```
+     */
+    template <typename T>
+    friend Quat<T> power(const Quat<T> &p, const Quat<T> &q, bool assumeUnit);
+
+    /**
+     * @brief return the value of power function with quaternion \f$q\f$.
+     * \f[p^q = e^{q\ln(p)}\f].
+     * @param q index quaternion of power function.
+     * @param assumeUnit if true, this quaternion should be a unit quaternion and this function will save some computations.
+     *
+     * For example
+     * ```
+     * Quatd p(1,2,3,4);
+     * Quatd q(5,6,7,8);
+     * p.power(q);
+     *
+     * p = p.normalize();
+     * p.power(q, true); //true means p is a unit quaternion
+     * ```
+     */
+    Quat<_Tp> power(const Quat<_Tp> &q, bool assumeUnit=false) const;
+
+    /**
+     * @brief return the crossProduct between \f$p = (a, b, c, d) = (a, \boldsymbol{u})\f$ and \f$q = (w, x, y, z) = (w, \boldsymbol{v})\f$.
+     * \f[p \times q = \frac{pq- qp}{2}\f].
+     * \f[p \times q = \boldsymbol{u} \times \boldsymbol{v}\f].
+     * \f[p \times q = (cz-dy)i + (dx-bz)j + (by-xc)k \f].
+     *
+     * For example
+     * ```
+     * Quatd q{1,2,3,4};
+     * Quatd p{5,6,7,8};
+     * crossProduct(p, q)
+     * ```
+     */
+    template <typename T>
+    friend Quat<T> crossProduct(const Quat<T> &p, const Quat<T> &q);
+
+    /**
+     * @brief return the crossProduct between \f$p = (a, b, c, d) = (a, \boldsymbol{u})\f$ and \f$q = (w, x, y, z) = (w, \boldsymbol{v})\f$.
+     * \f[p \times q = \frac{pq- qp}{2}\f].
+     * \f[p \times q = \boldsymbol{u} \times \boldsymbol{v}\f].
+     * \f[p \times q = (cz-dy)i + (dx-bz)j + (by-xc)k \f].
+     *
+     * For example
+     * ```
+     * Quatd q{1,2,3,4};
+     * Quatd p{5,6,7,8};
+     * p.crossProduct(q)
+     * ```
+     */
+    Quat<_Tp> crossProduct(const Quat<_Tp> &q) const;
+
+    /**
+     * @brief return the norm of quaternion.
+     * \f[||q|| = \sqrt{w^2 + x^2 + y^2 + z^2}\f].
+     */
+    _Tp norm() const;
+
+    /**
+     * @brief return a normalized \f$p\f$.
+     * \f[p = \frac{q}{||q||}\f]
+     * where \f$p\f$ satisfies \f$(p.x)^2 + (p.y)^2 + (p.z)^2 + (p.w)^2 = 1.\f$.
+     */
+    Quat<_Tp> normalize() const;
```

Comment:
I think `normalized` has more sense here; `normalize` looks more like a method that changes its object

Reply:
I think what you mean is as follows:

```cpp
// change q and return None
q.normalize()

// q is not changed and return normalized q
q2 = q.normalized()
```

It is a bit strange using adjectives as the names of built-in member functions. And since this module only has one of the above member functions, I think it is enough to use the verb without causing confusion.